### PR TITLE
Fix nil interface != nil issue

### PR DESCRIPTION
**Pull request by schachmat**
TL;DR: https://glucn.medium.com/golang-an-interface-holding-a-nil-value-is-not-nil-bb151f472cc7

When `Init()` calls `setup()` and `setup()` fails, it would return a `nil` value of the type `*syslog.Writer`. Storing this inside an `io.Writer` interface type variable `il`, `wl`, or `el` inside `Init()` means that those variables will not be `nil` (because they are a valid `nil` value of the `io.Writer` interface). Thus the checks `if il != nil` etc. will always be true and the `nil` value `syslog.Writer` pointer will always be added to the `iLogs`, `wLogs`, `eLogs` lists. Because we're in a case where `setup()` returned an error, we'll attempt to log the error later on using the `io.MultiWriter(eLogs...)` which contains one `nil` value writer. This means the `syslog.(*Writer).Write()` call will fail with segmentation violation and lead to the program crashing.

With this commit, we change the interface of `setup()` to return `io.Writer` interface values directly which can be safely compared against `nil` and  thus the `nil` value writers are never added to the `io.MultiWriter`s.

### CODE REVIEW

**Comment by pkern** [comment]
With the same argument, we need the same logic in `logger_windows.go`, no? Technically both in `setup` and `newW` if I understand the issue correctly.

**Comment by schachmat** [comment]
Yeah, good catch! I found that we at least need the same fix in `setup()` of `logger_windows.go`, but technically we wouldn't need it in `newW()`, because `setup()` does not compare the returned writer from `newW()` (at least currently), but relies on the `error` instead. However, to be safe against future changes I've also updated the interface of `newW()` and made both `io.WriteCloser` instead of `io.Writer` (currently not needed, but may be nice in the future).

I don't have a Windows machine to confirm the `logger_windows.go` part is actually working as intended, but I've tried to mirror the situation [in the playground](https://play.golang.org/p/LWtPKfiP_-C) (just remove the `//` for testing the error case). If you think it would be safer to also confirm the code is WAI on windows, maybe @alexherrero can check?

**Comment by pkern** [comment]
The tests pass on Windows and the code change looks logical. I think that's good enough for me. I'll try to get GitHub Actions for Windows going after I get the current migration away from Travis in.

**Review by pkern — APPROVED** [review]

**Comment by pkern** [comment]
Looks like I've only got write access, not admin access over the repo and cannot override the failing test.
